### DISCUSSION TITLE
Disabling auto-spell mode for markdown

### DISCUSSION THREAD

**Discussion by mikker** [LazyVim/LazyVim#392] [Q&A] [answered]
Hi!

I'm struggling to figure out how to disable [Lazy's autocommand to spell check markdown](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/autocmds.lua#L62-L70).

What would be the best approach?

Thanks!

**Comment by folke** [comment]
Add `  vim.api.nvim_del_augroup_by_name("wrap_spell")` to your `lua/config/autocmds.lua`

**Reply by mikker** [reply]
Perfect 👌
Thank you for your work!

**Reply by fang2hou** [reply]
`vim.api.nvim_del_augroup_by_name("lazyvim_wrap_spell")` in 2025.

ref: https://github.com/LazyVim/LazyVim/blob/ec5981dfb1222c3bf246d9bcaa713d5cfa486fbd/lua/lazyvim/config/autocmds.lua#L4

**Comment by mikker** [comment]
Getting this after restarting nvim:

```
Failed loading config.autocmds

/Users/mikker/.config/nvim/lua/config/autocmds.lua:32: Vim:E367: No such group: "wrap_spell"
```

**Comment by mikker** [comment, marked as answer]
Add `vim.api.nvim_del_augroup_by_name("lazyvim_wrap_spell")` to your `lua/config/autocmds.lua`

(Thank you @folke)

**Comment by weakish** [comment]
Delete `lazyvim_wrap_spell` also removes the wrap option for Markdown files.
If you still want to wrap long lines in Markdown files, you can add it back:

```lua
-- Disable built-in spellchecking for Markdown
vim.api.nvim_del_augroup_by_name("lazyvim_wrap_spell")
vim.api.nvim_create_autocmd("FileType", {
  group = vim.api.nvim_create_augroup("lazyvim_user_markdown", { clear = true }),
  pattern = { "gitcommit", "markdown" },
  callback = function()
    vim.opt_local.wrap = true
  end,
})

```

**Comment by franklouwers** [comment]
I've tried the settings above, but I still get the auto-correct popup, and (more annoyingly: hitting enters takes the first suggestion). Anyway to debug where it's still coming from?

**Reply by dpetka2001** [reply]
There is a source for `nvim-cmp` for spelling, namely [cmp-spell](https://github.com/f3fora/cmp-spell). Is there any chance you're using that?

**Comment by Tenshock** [comment]
Any more suggestions as of 2025? I don't find the plugin or configuration responsible to word suggestions or markdown snippets.

I tried to disable the markdown plugin without success. Whenever I type a number, I have a markdown snippet to create a table, when I start to type any word, I have words suggestions (with a little "abc" before the suggestion). I don't understand from which configuration this come from. I think there is at least to configurations (for markdown specificities and for words) but I can't find the related configuration.

**Reply by fang2hou** [reply]
https://github.com/LazyVim/LazyVim/discussions/392#discussioncomment-12683657

**Reply by Tenshock** [reply]
Thank you! Finally my problem was my configuration with [Blink.cmp](https://github.com/Saghen/blink.cmp)!

For me, the solution was to disable the Text completion item kind, here is the thread with the snippet that helped me: https://github.com/Saghen/blink.cmp/discussions/613#discussioncomment-12028607

**Comment by xaocon** [comment]
In case anyone else comes here looking for help because of spell messages after trying the above you may want to look at disabling the `spell` capture. I haven't found the best way to handle it so far but this might be helpful:

``` lua
local query = vim.treesitter.query.get("markdown", "highlights")
query.query:disable_capture("spell")
```

**Reply by AuthorProxy** [reply]
Here is for all files and warp is working

```
vim.api.nvim_create_autocmd("FileType", {
  pattern = "*",
  callback = function()
    vim.opt_local.spell = false
  end,
})
```

I think it should be prohibited to enable spellcheck by default

**Comment by plum** [comment]
Let me be frank. I don't need or desire  spell checking pop ups on every word I type - not any.

I'll go back to plain  Nvim add m own Telkenasten plugins ,fzf and so forth. it's a shame because the Lazy IDE design as a suite is just what is commonly needed. But it's entirely unsuited to writing in English or any other language other than computer programming.

 I' m a native English speaker of English. That's unimportant ,  perhaps the least important  of all speakers of English. 
 People who wish to communicate in  English NOW, without a dictionary or the initiative  to  learn to communicate and make clear sense  take priority.  get in on the on get in the game, 'improve  their lives'  leaving their own people behind.  I don't buy that myself.
Masters of the language have their place.  Without a high standard in human language society will berk down badly. No  wonder confusion is a frequent word now.  And **_their_**  << HA! HA! .......not unkind because they insist upon  high standards. No such spell checker, or sentence corrector will actually work to help anyone improve at all, including students of foreign languages. Yes, it will block their progress. We can can  forget that effort in assistance, in all kindness.

Its not because the spell suggestions in blink along with Lazyvim's autocommand in markdown  are annoying, but because it makes markdown impossible. A huge waste of time, with _constant destructive edits_ to  to correct in addition to _visual hell._ It is quickly tiring and reduces work output to about 1/10 th.

After **_trying for two years_**  to disable spell checking pop ups in comp or blink, with fancy autocommands that never work,  and using LAzyvim for  longer,  This was my last effort. Non of the suggestions above work.  I  read up, employ  them, source my init.lua  try again .. etc. and meanwhile I am working daily on  markdown  documents with distaste. Does me no good. I'd be better of with an electron app.
Two years! I have ten years left to live at most.

 I persist with  Nvim for good reason. It is slick for document writing - by far the best imo and alongside,  for Linux deployment. Folding  in Markdown  makes NVIM ascend above the rest, but its versatily and snappiness too. I'm talking about writing clear and structured document writing in human language,  both instructive and, with legal consequence.   Documents built through note taking in the same, Nvim.  Cat >> 'this file'  if not open.

I stand to be corrected . I am interested to hear replies, including any from the developers, as to the purpose of LazyVim in respect to writing in human language.


John Plumridge

ps. During those last few years I had been able to disable nvim.comp 's spell suggestion, and had a few weeks peace. But now that's gone and unfathomable blink has appeared to attack everyone attempting to write in human language in LazyVim.

**Comment by Amiganer** [comment]
For me it came first in the last update.
What is with another way to solve this: colorscheme. Default will be tokyonight.nvim and and setting "@spell" without "underline".
